File: src/main/java/com/app/projettic/service/UserExtraService.java
package com.app.projettic.service;

import com.app.projettic.domain.UserExtra;
import com.app.projettic.service.dto.UserExtraDTO;

import java.util.List;
import java.util.Optional;

/**
 * Service Interface for managing {@link com.app.projettic.domain.UserExtra}.
 */
public interface UserExtraService {

    /**
     * Save a userExtra.
     *
     * @param userExtraDTO the entity to save.
     * @return the persisted entity.
     */
    UserExtraDTO save(UserExtraDTO userExtraDTO);

    /**
     * Get all the userExtras.
     *
     * @return the list of entities.
     */
    List<UserExtraDTO> findAll();

    /**
     * Get the "id" userExtra.
     *
     * @param id the id of the entity.
     * @return the entity.
     */
    Optional<UserExtraDTO> findOne(Long id);

    /**
     * Get the "id" userExtra.
     *
     * @param actif the actif of the userExtra.
     * @return the entity.
     */
    List<UserExtraDTO> findByActif(boolean actif);

    /**
     * Delete the "id" userExtra.
     *
     * @param id the id of the entity.
     */
    void delete(Long id);

    /**
     * Search for the userExtra corresponding to the query.
     *
     * @param query the query of the search.
     *
     * @return the list of entities.
     */
    List<UserExtraDTO> search(String query);

    /**
     * Get all user from groupe id
     * @param groupeId
     * @return array list
     */
    List<UserExtraDTO> findByGroupeId(Long groupeId);
}

File: src/main/webapp/app/entities/user-extra/user-extra.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpResponse } from '@angular/common/http';
import { Observable } from 'rxjs';

import { SERVER_API_URL } from 'app/app.constants';
import { createRequestOption, Search } from 'app/shared/util/request-util';
import { IUserExtra } from 'app/shared/model/user-extra.model';
import { TypeCursus } from 'app/shared/model/enumerations/type-cursus.model';

type EntityResponseType = HttpResponse<IUserExtra>;
type EntityArrayResponseType = HttpResponse<IUserExtra[]>;

@Injectable({ providedIn: 'root' })
export class UserExtraService {
  public resourceUrl = SERVER_API_URL + 'api/user-extras';
  public resourceSearchUrl = SERVER_API_URL + 'api/_search/user-extras';

  constructor(protected http: HttpClient) {}

  create(userExtra: IUserExtra): Observable<EntityResponseType> {
    return this.http.post<IUserExtra>(this.resourceUrl, userExtra, { observe: 'response' });
  }

  update(userExtra: IUserExtra): Observable<EntityResponseType> {
    return this.http.put<IUserExtra>(this.resourceUrl, userExtra, { observe: 'response' });
  }

  find(id: number): Observable<EntityResponseType> {
    return this.http.get<IUserExtra>(`${this.resourceUrl}/${id}`, { observe: 'response' });
  }

  findAll(): Observable<IUserExtra[]> {
    return this.http.get<IUserExtra[]>(`${this.resourceUrl}`);
  }

  findByActif(actif: boolean): Observable<EntityArrayResponseType> {
    return this.http.get<IUserExtra[]>(`${this.resourceUrl}/actif/${actif}`, { observe: 'response' });
  }

  findByActifAndCursus(actif: boolean, cursus: TypeCursus): Observable<EntityArrayResponseType> {
    if(cursus !== null) {
      return this.http.get<IUserExtra[]>(`${this.resourceUrl}/cursus/${actif}/${cursus}`, { observe: 'response' });
    }else {
      return null;
    }
  }

  query(req?: any): Observable<EntityArrayResponseType> {
    const options = createRequestOption(req);
    return this.http.get<IUserExtra[]>(this.resourceUrl, { params: options, observe: 'response' });
  }

  delete(id: number): Observable<HttpResponse<{}>> {
    return this.http.delete(`${this.resourceUrl}/${id}`, { observe: 'response' });
  }

  search(req: Search): Observable<EntityArrayResponseType> {
    const options = createRequestOption(req);
    return this.http.get<IUserExtra[]>(this.resourceSearchUrl, { params: options, observe: 'response' });
  }

  findByGroupeId(groupeId: number): Observable<HttpResponse<IUserExtra[]>> {
    return this.http.get<IUserExtra[]>(`${this.resourceUrl}/groupe/${groupeId}`, { observe: 'response' });
  }
}
